composer_multi_project_monitoring/main.tf

```diff
@@ -0,0 +1,1018 @@
+# [START composer_multi_project_monitoring]
+#   Monitoring for multiple Cloud Composer environments
+#   
+#   Usage:
+#       1. Create a new project that you will use for monitoring of Cloud Composer environments in other projects
+#       2. Store the name of this project in "monitoring_project" variable below, in the locals block
+#       3. Save the list of projects with Cloud Composer environments to be monitored in "monitored_projects", set in the locals block
+#       4. Run "terraform apply"
+#
+#   The script creates the following resources in the monitoring project:
+#           1. Adds monitored projects to Cloud Monitoring
+#           2. Creates Alert Policies
+#           3. Creates Monitoring Dashboard
+#
+
+
+#######################################################
+#  
+# Overall settings
+#
+########################################################
+
+
+locals {
+    monitoring_project = "YOUR_MONITORING_PROJECT"
+    monitored_projects = toset(["YOUR_PROJECT_TO_MONITOR_1", "YOUR_PROJECT_TO_MONITOR_2", "YOUR_PROJECT_TO_MONITOR_3"])
+}
+
+#######################################################
+#  
+# Provider
+#
+########################################################
+
+provider "google-beta" {
+  project = local.monitoring_project
+}
+provider "google" {
+  project = local.monitoring_project
+}
+
+#######################################################
+#  
+# Add Monitored Projects to the Monitoring project
+#
+########################################################
+
+resource "google_monitoring_monitored_project" "projects_monitored" {
+  for_each = local.monitored_projects
+  metrics_scope = join("",["locations/global/metricsScopes/",local.monitoring_project])
+  name          = "${each.value}"
+  provider      = google-beta
+}
+
+
+#######################################################
+#  
+# Create alert policies in Monitoring project
+#
+########################################################
+
+resource "google_monitoring_alert_policy" "environment_health" {
+  display_name = "Environment Health"
+  combiner     = "OR"
+  conditions {
+    display_name = "Environment Health"
+    condition_monitoring_query_language {
+        query = join("", [
```

Comment:
Would a multiline string be more easier than the join func?

Reply:
No idea. It works as is, so I'm kind of of the mindset that if it's not broken I don't want to fix it, but if it's specifically an anti-pattern, I can try refactoring all ~25 to multiline strings to see if it works. Wdyt, @bharathkkb ?

Reply:
@leahecole I am unfamiliar with MQL and how users may want to consume it. We could iterate if we get feedback later.